### DISCUSSION TITLE
How to reuse existing json translations which have different syntax for pluralization

### DISCUSSION THREAD

**Discussion by ahnpnl** [robisim74/qwik-speak#40] [Ideas]
Hi,

I've recently thought about how to reuse existing JSON translation files. The scenario is my work has 1 legacy project which has some translations and we would like reuse those for the new project with Qwik.

The problem is that, the existing JSON translation uses `|` for pluralization, for example `banana: 'no bananas | {n} banana | {n} bananas'` and `qwik-speak` doesn't know this way.

Is it an idea that `qwik-speak` can provide option to define custom plural rules? For example similar like this https://vue-i18n.intlify.dev/guide/essentials/pluralization.html#custom-pluralization

Or any kinds of ways that we can use the same translation files for different projects like this case.

**Comment by robisim74** [comment]
I can't think of supporting such specific solutions for backwards compatibility with different implementations of different frameworks and libraries in a project like this.
Similar solutions involved the hydration of a lot of javascript code evaluated at runtime, while this project tries to do the translation at compile time.

I think you could solve the problem by implementing your own translate/plural function :

```
export function MyPlural(keys: string, params: any) {
  const ctx = useSpeakContext();
  const { locale, translation, config } = ctx;
 
  // Get the rule for the locale

  // Get the value in translation

  // Transform the value according to the rule

  return ...
}
```

However this might only work at runtime in any case, so you should put these translations in `runtimeAssets`.

**Reply by ahnpnl** [reply]
thanks :)
